mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/AnnotationScanner.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.util;
+
+import java.lang.annotation.Annotation;
+import java.util.Optional;
+import java.util.Set;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public class AnnotationScanner {
+  public static <A extends Annotation> Optional<ResolvedAnnotation<A>> getClassAnnotation(
+      Class<A> annotationType, Set<TypeElement> typeHierarchy) {
+    for (TypeElement element : typeHierarchy) {
+      A annotation = element.getAnnotation(annotationType);
+      if (annotation != null) {
+        return Optional.of(new ResolvedAnnotation<>(annotation, element));
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static <A extends Annotation> Optional<ResolvedAnnotation<A>> getMethodAnnotation(
+      Class<A> annotationType, ExecutableElement getMethod, Set<TypeElement> typeHierarchy) {
+    // first try evaluating the method as it is.
+    A annotation = getMethod.getAnnotation(annotationType);
+    if (annotation != null) {
+      return Optional.of(new ResolvedAnnotation<>(annotation, getMethod));
+    }
+
+    // otherwise navigate the hierarchy until an annotation is found.
+    for (TypeElement typeElement : typeHierarchy) {
+      for (Element child : typeElement.getEnclosedElements()) {
+        Set<Modifier> modifiers = child.getModifiers();
+        if (child.getKind() != ElementKind.METHOD
+            || modifiers.contains(Modifier.STATIC)
+            || modifiers.contains(Modifier.PRIVATE)) {
```

Comment:
So if someone by mistake annotates private method we will just ignore that? Maybe some kind of log stating that we are not taking private methods into accounts?

Reply:
I think it's fine to quietly ignore these as we did the same with the previous mapper, we do the same for identifying property methods.

In addition, I don't think it can reasonably happen, you can't override a parent method that is private.  We could possibly even remove the check for static & private, but I think it's fine to be paranoid.